### PR TITLE
Add dump memory command to interactive mode

### PR DESCRIPTION
**Pull request by jerryz123**

### CODE REVIEW

**Review by aswaterman — APPROVED** [review]
I'm fine with this.  Out of curiosity, what's the use case?

Any thoughts, @scottj97?

File: riscv/interactive.cc [hunk 1]
```diff
@@ -774,3 +776,16 @@ void sim_t::interactive_until(const std::string& cmd, const std::vector<std::str
...
+void sim_t::interactive_dumpmems(const std::string& cmd, const std::vector<std::string>& args)
+{
+  for (unsigned i = 0; i < mems.size(); i++) {
+    std::stringstream mem_fname;
+    mem_fname << "mem.0x" << std::hex << mems[i].first << ".bin";
+
+    std::ofstream mem_file;
+    mem_file.open(mem_fname.str());
+    mems[i].second->dump(mem_file);
+    mem_file.close();
```
Comment (scottj97) [inline riscv/interactive.cc]: Pass in the filename to the constructor, which will open the file. Then let the destructor do the close.

**Review by scottj97 — DISMISSED** [review]

**Comment by scottj97** [comment]
> I'm fine with this. Out of curiosity, what's the use case?
> 
> Any thoughts, @scottj97?

🤷‍♂️

**Comment by jerryz123** [comment]
> I'm fine with this. Out of curiosity, what's the use case?

This makes it easier to generate an architectural checkpoint from spike execution.

**Comment by aswaterman** [comment]
Makes sense.

**Pull request merged**